Lib/statistics.py

```diff
@@ -304,6 +304,19 @@ def _fail_neg(values, errmsg='negative value'):
             raise StatisticsError(errmsg)
         yield x
 
+def _isqrt_frac_rto(n: int, m: int) -> float:
+    'Square root of n/m, rounded to the nearest integer using round-to-odd.'
+    a = math.isqrt(n*m) // m
+    return a | (a*a*m != n)
+
+def _sqrt_frac(n: int, m: int) -> float:
```

Comment:
Note: this code was intended for the case n/m positive; I should have made that clearer in the issue discussion and the docstring. The sketch of correctness proof doesn't apply directly to the case n/m = 0. It does actually happen to work for the case n/m = 0, but mostly by accident, and a bit more inefficiently than necessary.

It's probably worth at least a test case for `_sqrt_frac(0, 1)`. For efficiency, we _could_ add a test and early return for the special case `n = 0`, but that just speeds up a presumably rather rare case at the expense of slowing down the common case. I leave that up to you - I'm happy either way.

Also worth noting that if `n/m` is negative, we _do_ get the expected `ValueError`. I'm not sure whether that's also worth a test.

Reply:
I'll add some test for the corner cases and for error handling.